src/btree/bt_ret.c

```diff
@@ -234,10 +230,8 @@ __wt_value_return_buf(WT_CURSOR_BTREE *cbt, WT_REF *ref, WT_ITEM *buf, WT_TIME_W
         if (tw != NULL)
             WT_TIME_WINDOW_COPY(tw, &unpack.tw);
         ret = __wt_page_cell_data_ref(session, page, &unpack, buf);
-#ifdef HAVE_DIAGNOSTIC
         if (ret == 0 && unpack.cell != NULL && is_ovfl_rm != NULL)
             *is_ovfl_rm = __wt_cell_type_raw(unpack.cell) == WT_CELL_VALUE_OVFL_RM;
```

Comment:
I've got two concerns with exposing this more broadly:
1) It's a corner case about overflow remove that is being exposed out of a general function buffer management. It makes it seem as though the layering isn't right. i.e: why do all callers care about removed overflow values, when they are trying to get an on-page reference?
2) I think the code had the complex ifdef HAVE_DIAGNOSTIC for performance reasons. I wonder if those performance reasons have gone away now? I vaguely recall that this code only needed to unpack the cell within the diagnostic logic before, but that seems to have changed. Could you check in on that? I'd like to make sure that if we are doing something that will change performance, we do it deliberately.

Reply:
It is me adding the diagnostic flag in WT-9935 to help debug this issue.

Reply:
I agree with the first point. I'm not sure how the whole overflow mechanism should work and when it is safe to remove them. I need to explore that a bit.

Reply:
I changed __wt_value_return_buf to return not found error if it reads an overflow removed cell instead of setting a buffer. The upper layer only needs to retry to handle this error. Hopefully this makes the api cleaner and layering more natural.

Reply:
This seems not working as `__wt_page_cell_data_ref` is called in many places.